### PR TITLE
Properly handle disabling of the LSP package

### LINKED ISSUE #2082: Output panels are not (reliably) destroyed if LSP is disabled
**Describe the bug**

"Diagnostics" and sometimes "LSP Server Log" output panels are still present after disabling LSP.

**To Reproduce**
1. Add LSP to "ignored_packages" in _Preferences.sublime-settings_ or run "Package Control: Disable Packgage"

**Expected behavior**

No LSP related output panel being left.

**Environment (please complete the following information):**
- OS: Windows 11
- Sublime Text version: 4137
- LSP version: master
- Language servers used: pyright, json, css, html, lemminx, ...

### PR DESCRIPTION
**Pull request by rchl**
Trying to better handle cleanup on disabling the LSP package and kill some leaks:
 - [x] Change `WindowRegistry.lookup` return value from `WindowManager` to `Optional[WindowManager]`. This ensures that when the LSP package gets disabled, existing references doing a lookup won't be able to create new instances of `WindowManager`.
 - [x] Ensure various functions in `panel.py` go through `WindowManager` when wanting to create output panel for `Window` so that they don't create panels when package is disabled.
 - [x] Don't unload global settings (`globalprefs`) on `plugin_unloaded` as that can cause crashes in code that is still running
 - [x] Create dummy `userpref` on `plugin_unloaded` so that accessing settings doesn't crash
 - [x] Remove `ConfigManager` and make `WindowRegistry` responsible for creating `WindowConfigManager`s and passing it to `WindowManager`. 
 - [x] Do something about `configs` ~~and (?) `client_configs`~~ globals. Maybe those should live and be controlled through `WindowRegistry` so that we have least amount of globals possible.

~~Note that due to ST bug https://github.com/sublimehq/sublime_text/issues/5148, we don't currently remove and delete instances of `WindowManager` so added a workaround for that so that we remove no longer valid instances now and then.~~ The bug was fixed.

Should fix #2082 when completed.

### CODE REVIEW

**Comment by rchl** [comment]
For now I'm done. I did some refactorings that should make it behave better when package is disabled/unloaded. Don't think everything is perfect yet but it should be getting there.

**Comment by rchl** [comment]
Actually I've just tested disabling the LSP package and it didn't really address the issue with panels not being destroyed. Or, they are destroyed but to async nature they come back immediately when last requests are being handled. Since now all panels are controlled through same path (PanelManager) it's easier to prevent that but a bit hesitant to add many repetitive checks...

File: plugin/configuration.py [hunk 1]
```diff
@@ -13,52 +15,60 @@ def run(self) -> None:
...
-            self.window.show_quick_panel(self._items, self._on_done)
+            self.window.show_quick_panel(self._items, partial(self._on_done, wm))
         else:
             self.window.status_message("No config available to disable")
 
-    def _on_done(self, index: int) -> None:
-        if index > -1:
-            config_name = self._items[index]
-            client_configs.disable(config_name)
+    def _on_done(self, wm: WindowManager, index: int) -> None:
```
Comment (predragnikolic) [inline plugin/configuration.py:52]: wm is unused here

File: plugin/core/settings.py [hunk 2]
```diff
@@ -132,30 +126,29 @@ def load_settings() -> None:
...
     if _settings_obj is not None:
-        _global_settings = None
         _settings_registration = None
-        _settings_obj = None
+        _settings_obj = sublime.load_settings("")
+        _settings = Settings(_settings_obj)
 
 
 def userprefs() -> Settings:
-    global _settings
```
[the remark this answers is not in the document]
Comment (rchl) [reply plugin/core/settings.py:153]: It's only necessary to override the value of the variable, not to read it.

**Comment by rchl** [comment]
Think nobody wants to waste their live reviewing it properly so life will review it.

**Comment by predragnikolic** [comment]
> Think nobody wants to waste their life reviewing it properly so life will review it.

pretty much like the code changes in this PR but haven't tested it  :)

**Pull request merged**